packages/types/surveys/types.ts

```diff
@@ -489,48 +547,381 @@ export const ZSurveyInlineTriggers = z.object({
-  welcomeCard: ZSurveyWelcomeCard,
-  questions: ZSurveyQuestions,
-  thankYouCard: ZSurveyThankYouCard,
-  hiddenFields: ZSurveyHiddenFields,
-  delay: z.number(),
-  autoComplete: z.number().nullable(),
-  runOnDate: z.date().nullable(),
-  closeOnDate: z.date().nullable(),
-  productOverwrites: ZSurveyProductOverwrites.nullable(),
-  styling: ZSurveyStyling.nullable(),
-  surveyClosedMessage: ZSurveyClosedMessage.nullable(),
-  segment: ZSegment.nullable(),
-  singleUse: ZSurveySingleUse.nullable(),
-  verifyEmail: ZSurveyVerifyEmail.nullable(),
-  pin: z.string().nullish(),
-  resultShareKey: z.string().nullable(),
-  displayPercentage: z.number().min(0.01).max(100).nullable(),
-  languages: z.array(ZSurveyLanguage),
-  showLanguageSwitch: z.boolean().nullable(),
-});
-
-export const ZSurveyUpdateInput = ZSurvey.omit({ createdAt: true, updatedAt: true }).and(
-  z.object({
-    createdAt: z.coerce.date(),
-    updatedAt: z.coerce.date(),
+export const ZSurvey = z
+  .object({
+    id: z.string().cuid2(),
+    createdAt: z.date(),
+    updatedAt: z.date(),
+    name: z.string(),
+    type: ZSurveyType,
+    environmentId: z.string(),
+    createdBy: z.string().nullable(),
+    status: ZSurveyStatus,
+    displayOption: ZSurveyDisplayOption,
+    autoClose: z.number().nullable(),
+    triggers: z.array(z.object({ actionClass: ZActionClass })),
+    redirectUrl: z.string().url({ message: "Invalid redirect URL" }).nullable(),
+    recontactDays: z.number().nullable(),
+    displayLimit: z.number().nullable(),
+    welcomeCard: ZSurveyWelcomeCard,
+    questions: ZSurveyQuestions.min(1, {
+      message: "Survey must have at least one question",
+    }).superRefine((questions, ctx) => {
+      const questionIds = questions.map((q) => q.id);
+      const uniqueQuestionIds = new Set(questionIds);
+      if (uniqueQuestionIds.size !== questionIds.length) {
+        ctx.addIssue({
+          code: z.ZodIssueCode.custom,
+          message: "Question IDs must be unique",
+          path: [questionIds.findIndex((id, index) => questionIds.indexOf(id) !== index), "id"],
+        });
+      }
+    }),
+    thankYouCard: ZSurveyThankYouCard,
+    hiddenFields: ZSurveyHiddenFields,
+    delay: z.number(),
+    autoComplete: z.number().nullable(),
+    runOnDate: z.date().nullable(),
+    closeOnDate: z.date().nullable(),
+    productOverwrites: ZSurveyProductOverwrites.nullable(),
+    styling: ZSurveyStyling.nullable(),
+    showLanguageSwitch: z.boolean().nullable(),
+    surveyClosedMessage: ZSurveyClosedMessage.nullable(),
+    segment: ZSegment.nullable(),
+    singleUse: ZSurveySingleUse.nullable(),
+    verifyEmail: ZSurveyVerifyEmail.nullable(),
+    pin: z.string().min(4, { message: "PIN must be a four digit number" }).nullish(),
+    resultShareKey: z.string().nullable(),
+    displayPercentage: z.number().min(0.01).max(100).nullable(),
+    languages: z.array(ZSurveyLanguage),
   })
-);
+  .superRefine((survey, ctx) => {
```

Comment:
Please add a few comments in the superRefine handle where you think clarification is necessary.